cms/static/js/views/settings/main.js

```diff
@@ -164,18 +174,41 @@ var DetailsView = ValidatingView.extend({
         $(e.currentTarget).attr('title', currentTimeText);
     },
     updateModel: function(event) {
+        var url, image_name;
         switch (event.currentTarget.id) {
         case 'course-image-url':
             this.setField(event);
-            var url = $(event.currentTarget).val();
-            var image_name = _.last(url.split('/'));
+            url = $(event.currentTarget).val();
+            image_name = _.last(url.split('/'));
             this.model.set('course_image_name', image_name);
             // Wait to set the image src until the user stops typing
             clearTimeout(this.imageTimer);
             this.imageTimer = setTimeout(function() {
                 $('#course-image').attr('src', $(event.currentTarget).val());
             }, 1000);
             break;
+        case 'hero-image-url':
```

Comment:
It is possible that course-image-url does not have test coverage. I was hoping for unit test coverage in main_spec.js, or if that is not practical or useful, a Bok Choy test that verifies that the behavior is correct (the uploaded image is correctly stored and displayed).

Note that there is also duplicated code here that should be pulled into a helper method.